canine/localization/file_handlers.py

```diff
@@ -393,34 +413,50 @@ def __init__(self, path, **kwargs):
         self.prefix = url_parse[1]
         self.uuid = url_parse[2]
 
-        # the actual filename is encoded in the content-disposition header;
-        # save this to self.path
-        # since the filesize and hashes are also encoded in the header, populate
-        # these fields now
-        resp_headers = subprocess.run(
-          'curl -s -D - -o /dev/full {token_flag} {file}'.format(
-            token_flag = self.token_flag,
-            file = self.path
-          ),
-          shell = True,
-          capture_output = True
-        )
         try:
-            headers = pd.DataFrame(
-              [x.split(": ") for x in resp_headers.stdout.decode().split("\r\n")[1:]],
-              columns=["header", "value"],
-            ).set_index("header")["value"]
-
-            self.path = re.match(".*filename=(.*)$", headers["Content-Disposition"])[1]
-            self._size = int(headers["Content-Length"])
-            self._hash = headers["Content-MD5"]
+            self.uri = type(self).gdc_drs_root + self.uuid
+            self.drs_obj = HandleDRSURI(self.uri, **self.extra_args)
```

Comment:
Is there a better way to handle the logic of checking if a drs:// URL exists for a GDC HTTP URL besides `try`/`catch`?

Reply:
That's the Python way! It takes advantage of the fact that a straight call to DRS will fail if it doesn't exist.